src/lib/hooks/swap/useSwapInfo.tsx

```diff
@@ -95,23 +104,15 @@ function useComputeSwapInfo(): SwapInfo {
       impact,
       inputUSDC,
       outputUSDC,
+      setSwapInfo,
       slippage,
       trade,
     ]
   )
 }
 
-const swapInfoAtom = atom<SwapInfo>({
-  [Field.INPUT]: {},
-  [Field.OUTPUT]: {},
-  trade: { state: TradeState.INVALID },
-  slippage: { auto: true, allowed: new Percent(0) },
-})
-
 export function SwapInfoUpdater() {
```

Comment:
hm why are we using a swap info atom here instead of just using the return value? that hook shouldn't be called from multiple places?